### DISCUSSION TITLE
How to Train And Test FasterRCNN?

### DISCUSSION THREAD

**Discussion by SamihaSara** [facebookresearch/detectron2#2850] [Q&A]
I am trying to train MaskRCNN on MOT20 dataset which does not provide polygon or segmentation masks. Can I train without Mask Head/Instance Segmentation? Please suggest how to do it.

**Comment by geep33** [comment]
Hiw an i supposed to know

**Comment by yogeshkumarpilli** [comment]
@SamihaSara  You can train fasterrcnn on your model. If you have correct annotations as per COCO format

**Reply by SamihaSara** [reply]
Can you tell me how to add another regression head to regress a feature (value ranges from 0 to 1)to the existing FasterRCNN model?

**Reply by yogeshkumarpilli** [reply]
Hello, Firstly as per your query. You cannot train the MASKRCNN without polygon annotations. If you have already the Annotations with only Bounding Boxes you can do an object detection task. 
Secondly, Regarding your other regression head query.
Do you want to modify the model of FasterRCNN? 
Can you be precise with your question

**Reply by SamihaSara** [reply]
Thanks for helping out. 
As I do not have polygon or mask annotation, I am now trying to Run FasterRCNN on the MOT20 dataset. But I want to add an extra feature in Dataset Annotation then Train FasterRCNN to regress that value. 
This feature is similar to the confidence score and can be calculated based on the detected objects. Since this is a new feature, I will have to modify some parts of FasterRCNN. 
Could you please suggest how can I extract features from the detected objects in a frame and find values of a custom added feature?
